cs-json/src/main/java/io/cloudslang/content/json/actions/AddPropertyToObject.java

```diff
@@ -0,0 +1,166 @@
+import com.hp.oo.sdk.content.annotations.Param;
+import com.hp.oo.sdk.content.annotations.Response;
+import com.hp.oo.sdk.content.plugin.ActionMetadata.MatchType;
+import com.hp.oo.sdk.content.plugin.ActionMetadata.ResponseType;
+import io.cloudslang.content.constants.OutputNames;
+import io.cloudslang.content.constants.ResponseNames;
+import io.cloudslang.content.constants.ReturnCodes;
+import io.cloudslang.content.json.utils.Constants;
+import io.cloudslang.content.utils.StringUtilities;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.cloudslang.content.json.utils.JsonUtils.populateResult;
+
+public class AddPropertyToObject {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public AddPropertyToObject() {
+        objectMapper.configure(JsonParser.Feature.ALLOW_SINGLE_QUOTES, true);
+    }
+
+    /**
+     * Inserts a new name/value property into a JSON object, where the value is a string.
+     * This operation forces the value of the name/vale pair to be a string, no matter what data type it actually is.
+     * This means that the <b>newPropertyValue</b> input does not need to be wrapped in quotes as any quotes entered in this
+     * input will be escaped, ending up in the properties' value.
+     * In case that a new property with the same name as an existing one is added, the old property's value will be overwritten.
+     * The 'Add JSON Property to Object' operation should be used to add values of type other than string,
+     * for example: another object, an array or a number.
+     *
+     * @param jsonObject       String representation of a JSON object. Objects in JSON are a collection of name value pairs,
+     *                         separated by a colon and surrounded with curly brackets {}. The name must be a string value,
+     *                         and the value can be a single string or any valid JSON object or array.
+     *                         Examples: {"one":1, "two":2}, {"one":{"a":"a","B":"B"}, "two":"two", "three":[1,2,3.4]}
+     * @param newPropertyName  The name of the new property to add to the JSON object. There is no rule as to which character to use.
+     *                         Examples: property1, some_property, another property
+     * @param newPropertyValue The value for the new property. This is interpreted as a string, no matter what the contents of the input.
+     *                         Examples: value, 1, [1,2,3]
+     * @return a map containing the output of the operation. Keys present in the map are:
+     * <p/>
+     * <br><br><b>returnResult</b> - This will contain the JSON with the new property/value added.
+     * <br><b>exception</b> - In case of success response, this result is empty. In case of failure response,
+     * this result contains the java stack trace of the runtime exception.
+     * <br><br><b>returnCode</b> - The returnCode of the operation: 0 for success, -1 for failure.
+     */
```

Comment:
you don't need the extra spacing here. Also, you don't really need the whole description here as you already have the descriptions on the operation, inputs and outputs.